parse.go

```diff
@@ -151,39 +162,25 @@ func (p *parser) parseComplexMessage() (ComplexMessage, error) {
 			return ComplexMessage{}, fmt.Errorf("got error token: '%s'", itm.val)
 		case itemWhitespace:
 			continue
-		case itemKeyword:
-			switch itm.val {
-			// TODO: case ReservedKeyword:
-			case "." + keywordInput: // Input declaration
-				p.next() // skip keyword
-
-				declaration, err := p.parseInputDeclaration()
-				if err != nil {
-					return ComplexMessage{}, fmt.Errorf("parse input declaration: %w", err)
-				}
-
-				declarations = append(declarations, declaration)
 
-			case "." + keywordLocal: // Local declaration
-				p.next() // skip keyword
-
-				declaration, err := p.parseLocalDeclaration()
-				if err != nil {
-					return ComplexMessage{}, fmt.Errorf("parse local declaration: %w", err)
-				}
+		case itemInputKeyword, itemLocalKeyword, itemReservedKeyword: // Declarations
```

Comment:
Why did you add `parseDeclaration()`? Instead. as you suggested per the ticket?

Reply:
because error handling makes it ugly
originlly when ticket was created error handling was not yet implemented, so
that would be simply
```go
switch itm.typ {
  case itemInputKeyword:
    declarations = append(declarations, p.parseInputDeclaration())
  case itemlocalKeyword:
    declarations = append(declarations, p.parseLocalDeclaration())
...
}
```

but now it would be

Reply:
what do you mean by ugly?

Reply:
by ugly, do you mean the `if` statement?

```go
case itemInputKeyword:
  v, err := p.parseInputKeyword()
  if err != nil {
  /// ...
  }
  
  declarations = append(declarations, v)

Reply:
yeah, for every declaration type

Reply:
i can change it no problem, but I just thought, that is a little bit cleaner that way

Reply:
I'm not convinced this PR improves the code:

* the code size increased significantly
* cognitive load increased as well

Is it worth still to pursue it? Should we postpone improvement until we figure out how to do it properly?

Reply:
code size increasement is mainly due to exhaustive linter. I would drop it as my experience it gives more headache than benefits :)

Otherwise apart from that, there is no significant code size changes.

About cognitive load, it decreased in one way, but maybe increased in other way, so I would say that is not changed much